framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -471,17 +505,17 @@ def localFinalizeActualSampling(self,jobObject,model,myInput):
             if self.convergeTraj[traj]:
               self.status[traj] = {'process':None, 'reason':'converged'}
             else:
-              # if rejecting bad point, keep the old point as the new point
-              if self.status[traj]['reason'] != 'rejecting bad opt point':
-                try:
-                  self.counter['recentOptHist'][traj][1] = copy.deepcopy(self.counter['recentOptHist'][traj][0])
-                except KeyError:
-                  # this means we don't have an entry for this trajectory yet, so don't copy anything
-                  pass
-                self.counter['recentOptHist'][traj][0] = {'inputs':self.optVarsHist[traj][self.counter['varsUpdate'][traj]],
-                                                          'output':currentObjectiveValue}
               # update status to submitting grad eval points
               self.status[traj]['process'] = 'submitting grad eval points'
+            # if rejecting bad point, keep the old point as the new point; otherwise, add the new one
+            if self.status[traj]['reason'] != 'rejecting bad opt point':
```

Comment:
Yes, Dan and I went a couple rounds with this trying to determine the best way to implement this.  We thought using an `enum` would work well, but even in python3 enums are slower than comparison to strings OR integers.  I think the findings are in the comments to the merge request #218.

While integer comparison IS a little bit faster than string comparison, it isn't nearly as human-friendly, which is the majority of the reason the "status" was implemented in the previous merge request.

It is definitely possible for me to revert the change, but I wasn't convinced by @maljovec findings that swapping to integers would be worthwhile.  I'm up for being convinced, though.